Fix for Move-Item with wildcard source and a non-existent destination directory implodes first directory

alexprabhat99 opened the pull request:
<!-- Anything that looks like this is a comment and can't be seen after the Pull Request is created. -->

# PR Summary

<!-- Summarize your PR between here and the checklist. -->
Fixes: https://github.com/PowerShell/PowerShell/issues/24837

## PR Context

<!-- Provide a little reasoning as to why this Pull Request helps and why you have opened it. -->
This PR fixes the follwing scenario:
When you call Move-Item with a wildcard into a directory that doesn't exist yet, it seems like the first move of a directory acts as a directory creation, which ends up moving the files into the new directory, instead of creating a new sub-directory, and I don't understand why / find this a rather odd behavior.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge. If this PR is a work in progress, please open this as a [Draft Pull Request and mark it as Ready to Review when it is ready to merge](https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests).
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [x] None
  - **OR**
  - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
    - [ ] Experimental feature name(s): <!-- Experimental feature name(s) here -->
- **User-facing changes**
  - [x] Not Applicable
  - **OR**
  - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed: <!-- Number/link of that issue here -->
- **Testing - New and feature**
  - [ ] N/A or can only be tested interactively
  - **OR**
  - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

alexprabhat99 committed:
Fix Move-Item wildcard moves to preserve directory structure

alexprabhat99 committed:
Added test for Move-Item to preserve directory structure

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1:
```diff
@@ -345,6 +345,41 @@ Describe "Basic FileSystem Provider Tests" -Tags "CI" {
             Copy-Item -Path $zeroLengthFile -Destination "$TestDrive\zeroLengthFile2.txt" -Force
             "$TestDrive\zeroLengthFile2.txt" | Should -Exist
          }
+
```
We never add references to issues.

iSazonov commented:
@alexprabhat99 Please look how Copilot fixed the issue in #26604. Copilot started with fix like yours and did some iterations to avoid side effects. If you agree with that fix you can grab it.

alexprabhat99 committed:
Fix Move-Item wildcard moves to preserve directory structure

alexprabhat99 commented:
Hi @iSazonov,
Sorry for the delay, I got caught up with a few things. I've updated the PR with the requested changes. Whenever you have a chance, I'd really appreciate it if you could take another look.
Thank you!

alexprabhat99 commented:
[Dongbo Wang (@daxian-dbw)](https://github.com/daxian-dbw) Could you please review this PR as potential fix for the issue: https://github.com/PowerShell/PowerShell/issues/24837? Thank you.

alexprabhat99 commented:
Hi @iSazonov, just a gentle reminder when you have a chance. I’ve addressed the review comments and updated the PR accordingly. Could you please take another look? Thanks!
